experimental/packages/opentelemetry-configuration/src/ConfigProvider.ts

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { DiagLogLevel } from '@opentelemetry/api';
+import {
+  ConfigurationModel,
+  initializeDefaultConfiguration,
+} from './configModel';
+import {
+  getBooleanFromEnv,
+  getStringFromEnv,
+  getStringListFromEnv,
+  diagLogLevelFromString,
+} from '@opentelemetry/core';
+
+/**
+ * ConfigProvider provides a configuration.
+ */
+export class ConfigProvider {
```

Comment:
I wonder if we could define an interface `ConfigProvider` and instead offer a `function getEnvironmentConfigProvider(): ConfigProvider {...}` factory to avoid the type problems outlined in #5283